pkg/api/v1beta3/types.go

```diff
@@ -820,12 +820,12 @@ type NodeSpec struct {
 
 // NodeStatus is information about the current status of a node.
 type NodeStatus struct {
-	// Queried from cloud provider, if available.
-	HostIP string `json:"hostIP,omitempty"`
 	// NodePhase is the current lifecycle phase of the node.
 	Phase NodePhase `json:"phase,omitempty"`
 	// Conditions is an array of current node conditions.
 	Conditions []NodeCondition `json:"conditions,omitempty"`
+	// Queried from cloud provider, if available.
+	Addresses []NodeAddress `json:"addresses,omitempty"`
```

Comment:
Are these addresses always observable from external sources? I want it to be a hard requirement that status be entirely reconstructable by observation. 

If not, we should put them in spec, even though they are populated automatically most of the time. It wouldn't be different from the other info in spec that the node controller posts.